shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/register/data/RegisterUserRepository.kt

```diff
@@ -0,0 +1,16 @@
+package com.vml.tutorial.plantshop.register.data
+
+import com.vml.tutorial.plantshop.profile.domain.User
+import com.vml.tutorial.plantshop.profile.domain.UserDataSource
+
+interface RegisterUserRepository {
+    suspend fun insertToDatabase(user: User)
+}
+
+class RegisterUserRepositoryImpl(
+    private val remoteDbUserDataSource: UserDataSource
+): RegisterUserRepository {
+    override suspend fun insertToDatabase(user: User) {
+        remoteDbUserDataSource.insertToDatabase(user)
+    }
+}
```

Comment:
I think you don't need an extra repository :) You can do all authentication-related stuff (login, register ) in authRepository. If you don't agree with me just ignore this comment :)

Reply:
I plan to do more work on it, so I might remove it later if it's not needed

Reply:
Okay